devlib/trace/ftrace.py

```diff
@@ -163,7 +161,19 @@ def __init__(self, target,
                     self.target.logger.warning(message)
                 else:
                     selected_functions.append(function)
-            self.function_string = _build_trace_functions(selected_functions)
+
+            if self.tracer is None:
+                self.function_string = _build_trace_functions(selected_functions)
+                # If function profiling is enabled we always need at least one event.
+                # Thus, if not other events have been specified, try to add at least
+                # a tracepoint which is always available and possibly triggered few
+                # times.
+                if not selected_events:
+                    selected_events = ['sched_wakeup_new']
+            elif self.tracer == 'function_graph':
+                self.function_string = _build_graph_functions(selected_functions, trace_children_functions)
```

Comment:
```
else:   # unknown  tracer
````

Reply:
There are quite a few other tracers, and they seem to work (haven't tried them all but a couple of them): `man trace-cmd-record` gives this list: function, function_graph, preemptirqsoff,  irqsoff, preemptoff and wakeup
but in practice the list of supported tracers depends on the kernel.

Reply:
Can we query the available tracers from debugfs and then match against that?

Reply:
Yes, do you want me to add some validation on the tracer ?

Reply:
That would be great.